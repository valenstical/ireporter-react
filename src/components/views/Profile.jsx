import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import loadingIcon from '../../../public/images/resources/loader-light.gif';
import Alert from './Alert';

/**
 * The edit profile component
 * @returns {object} The generated JSX object
 */
export default function Profile({ handleEditProfile, user }) {
  return (
    <main className="login-wrapper top-space" data-pg-name="Dashboard">
      <section className="heading">
        <div className="wrapper" data-pg-name="Wrapper">
          <div className="bread">
            <Link to="/dashboard">Home</Link>
            <i className="fa fa-angle-right" />
            <a className="active">Edit Profile</a>
          </div>
        </div>
      </section>
      <section className="edit-profile-wrapper">
        <div className="wrapper" data-pg-name="Wrapper">
          <div className="row">
            <div className="column column-md-8">
              <form className="card" onSubmit={handleEditProfile} noValidate>
                {user.message.length > 0 && <Alert message={user.success ? ['Your profile has was updated successfully.'] : user.message} title={user.success ? 'Profile Updated!' : 'Something went wrong!'} success={user.success} />}
                <div className="form-wrapper">
                  <label htmlFor="firstname">First Name </label>
                  <input className="form-element text-capitalize" type="text" name="firstname" required id="firstname" defaultValue={user.firstname} />
                </div>
                <div className="form-wrapper">
                  <label htmlFor="lastname">Last Name </label>
                  <input id="lastname" className="form-element text-capitalize" type="text" name="lastname" required defaultValue={user.lastname} />
                </div>
                <div className="form-wrapper">
                  <label htmlFor="othername">Other Name</label>
                  <input id="othername" className="form-element text-capitalize" type="text" name="othernames" defaultValue={user.othernames} />
                </div>
                <div className="form-wrapper">
                  <label htmlFor="email">Email </label>
                  <input id="email" className="form-element" type="email" name="email" required defaultValue={user.email} />
                </div>
                <div className="form-wrapper">
                  <label htmlFor="phone">Phone Number </label>
                  <input id="phone" className="form-element" type="tel" name="phoneNumber" required defaultValue={user.phoneNumber} />
                </div>
                <div className="form-wrapper">
                  <label htmlFor="username">Username </label>
                  <input id="username" className="form-element" type="text" name="username" required defaultValue={user.username} />
                </div>
                <button className="btn-brand btn-250 btn-submit" type="submit" disabled={user.isBusy}>
                  <span>Update Profile</span>
                  { user.isBusy && <img src={loadingIcon} width="12" alt="loading" />}
                </button>
                <input type="hidden" name="userId" value={user.id} />
              </form>
            </div>
          </div>
        </div>
      </section>
    </main>
  );
}

Profile.propTypes = {
  handleEditProfile: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired,
};
