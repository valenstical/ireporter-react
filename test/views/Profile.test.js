import React from 'react';
import { shallow } from 'enzyme';
import Profile from '../../src/components/views/Profile';

describe('<Profile>', () => {
  it('should render without crashing on successful profile update', () => {
    const wrapper = shallow(
      <Profile
      user={{ isBusy: true, success: true, message: ['success'] }}
      handleEditProfile={jest.fn()}

    />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing on failed profile update', () => {
    const wrapper = shallow(
      <Profile
      user={{ isBusy: false, success: false, message: ['failed'] }}
      handleEditProfile={jest.fn()}
    />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
