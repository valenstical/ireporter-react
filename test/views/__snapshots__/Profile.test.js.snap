// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile> should render without crashing on failed profile update 1`] = `
<main
  className="login-wrapper top-space"
  data-pg-name="Dashboard"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
      data-pg-name="Wrapper"
    >
      <div
        className="bread"
      >
        <Link
          replace={false}
          to="/dashboard"
        >
          Home
        </Link>
        <i
          className="fa fa-angle-right"
        />
        <a
          className="active"
        >
          Edit Profile
        </a>
      </div>
    </div>
  </section>
  <section
    className="edit-profile-wrapper"
  >
    <div
      className="wrapper"
      data-pg-name="Wrapper"
    >
      <div
        className="row"
      >
        <div
          className="column column-md-8"
        >
          <form
            className="card"
            noValidate={true}
            onSubmit={[MockFunction]}
          >
            <Alert
              message={
                Array [
                  "failed",
                ]
              }
              success={false}
              title="Something went wrong!"
            />
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="firstname"
              >
                First Name 
              </label>
              <input
                className="form-element text-capitalize"
                id="firstname"
                name="firstname"
                required={true}
                type="text"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="lastname"
              >
                Last Name 
              </label>
              <input
                className="form-element text-capitalize"
                id="lastname"
                name="lastname"
                required={true}
                type="text"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="othername"
              >
                Other Name
              </label>
              <input
                className="form-element text-capitalize"
                id="othername"
                name="othernames"
                type="text"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="email"
              >
                Email 
              </label>
              <input
                className="form-element"
                id="email"
                name="email"
                required={true}
                type="email"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="phone"
              >
                Phone Number 
              </label>
              <input
                className="form-element"
                id="phone"
                name="phoneNumber"
                required={true}
                type="tel"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="username"
              >
                Username 
              </label>
              <input
                className="form-element"
                id="username"
                name="username"
                required={true}
                type="text"
              />
            </div>
            <button
              className="btn-brand btn-250 btn-submit"
              disabled={false}
              type="submit"
            >
              <span>
                Update Profile
              </span>
            </button>
            <input
              name="userId"
              type="hidden"
            />
          </form>
        </div>
      </div>
    </div>
  </section>
</main>
`;

exports[`<Profile> should render without crashing on successful profile update 1`] = `
<main
  className="login-wrapper top-space"
  data-pg-name="Dashboard"
>
  <section
    className="heading"
  >
    <div
      className="wrapper"
      data-pg-name="Wrapper"
    >
      <div
        className="bread"
      >
        <Link
          replace={false}
          to="/dashboard"
        >
          Home
        </Link>
        <i
          className="fa fa-angle-right"
        />
        <a
          className="active"
        >
          Edit Profile
        </a>
      </div>
    </div>
  </section>
  <section
    className="edit-profile-wrapper"
  >
    <div
      className="wrapper"
      data-pg-name="Wrapper"
    >
      <div
        className="row"
      >
        <div
          className="column column-md-8"
        >
          <form
            className="card"
            noValidate={true}
            onSubmit={[MockFunction]}
          >
            <Alert
              message={
                Array [
                  "Your profile has was updated successfully.",
                ]
              }
              success={true}
              title="Profile Updated!"
            />
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="firstname"
              >
                First Name 
              </label>
              <input
                className="form-element text-capitalize"
                id="firstname"
                name="firstname"
                required={true}
                type="text"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="lastname"
              >
                Last Name 
              </label>
              <input
                className="form-element text-capitalize"
                id="lastname"
                name="lastname"
                required={true}
                type="text"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="othername"
              >
                Other Name
              </label>
              <input
                className="form-element text-capitalize"
                id="othername"
                name="othernames"
                type="text"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="email"
              >
                Email 
              </label>
              <input
                className="form-element"
                id="email"
                name="email"
                required={true}
                type="email"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="phone"
              >
                Phone Number 
              </label>
              <input
                className="form-element"
                id="phone"
                name="phoneNumber"
                required={true}
                type="tel"
              />
            </div>
            <div
              className="form-wrapper"
            >
              <label
                htmlFor="username"
              >
                Username 
              </label>
              <input
                className="form-element"
                id="username"
                name="username"
                required={true}
                type="text"
              />
            </div>
            <button
              className="btn-brand btn-250 btn-submit"
              disabled={true}
              type="submit"
            >
              <span>
                Update Profile
              </span>
              <img
                alt="loading"
                src={Object {}}
                width="12"
              />
            </button>
            <input
              name="userId"
              type="hidden"
            />
          </form>
        </div>
      </div>
    </div>
  </section>
</main>
`;
